streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -88,7 +90,8 @@ public void process(final K key, final V1 value) {
             final long timeFrom = Math.max(0L, context().timestamp() - joinBeforeMs);
             final long timeTo = Math.max(0L, context().timestamp() + joinAfterMs);
 
-            try (final WindowStoreIterator<V2> iter = otherWindow.fetch(key, timeFrom, timeTo)) {
+            try (final WindowStoreIterator<V2> iter =
+                     otherWindow.fetch(key, Instant.ofEpochMilli(timeFrom), Duration.ofMillis(timeTo - timeFrom))) {
```

Comment:
I  thought we agreed to add `WindowStoreIterator<V> fetch(K key, long timeFrom, long timeTo);` to the `WindowStore` interface, so that we could continue to call the long variant internally without a deprecation warning.

Reply:
This would also apply to the other methods that are now deprecated on the `ReadOnly` stores.